python/amici/petab_import.py

```diff
@@ -45,43 +47,42 @@ def get_fixed_parameters(
         List of IDs of parameters which are to be considered constant.
     """
 
-    # column names are model parameter names that should be made constant
-    # except for any overridden parameters
+    # Column names are model parameter IDs, compartment IDs or speciesIDs.
+    # Thereof, all parameters except for any overridden ones should be made
+    # constant.
     # (Could potentially still be made constant, but leaving them might
     # increase model reusability)
+
+    # handle parameters in condition table
     if condition_df is not None:
         fixed_parameters = list(condition_df.columns)
+        # get rid of conditionName column
         try:
-            fixed_parameters.remove('conditionName')
+            fixed_parameters.remove(CONDITION_NAME)
         except ValueError:
             pass
-        # remove overridden parameters
+
+        # remove overridden parameters (`object`-type columns)
         fixed_parameters = [p for p in fixed_parameters
-                            if condition_df[p].dtype != 'O']
+                            if condition_df[p].dtype != 'O'
+                            and sbml_model.getParameter(p) is not None]
         # must be unique
         if len(fixed_parameters) != len(set(fixed_parameters)):
             raise AssertionError(
                 'len(fixed_parameters) != len(set(fixed_parameters))')
     else:
         fixed_parameters = []
 
-    # States occurring as column names of the condition table need to be
-    #  converted to parameters
-    # TODO https://github.com/ICB-DCM/PEtab/issues/181
-    species_to_convert = [x for x in fixed_parameters
-                          if sbml_model.getSpecies(x)]
-    species_to_parameters(species_to_convert, sbml_model)
-
     # Others are optional
     if const_species_to_parameters:
         # Turn species which are marked constant in the SBML model into
         # parameters
         constant_species = constant_species_to_parameters(sbml_model)
 
-        logger.log(logging.INFO, "Constant species converted to parameters "
-                   + str(len(constant_species)))
-        logger.log(logging.INFO, "Non-constant species "
-                   + str(len(sbml_model.getListOfSpecies())))
+        logger.info("Constant species converted to parameters "
```

Comment:
```suggestion
        logger.info("Constant species converted to parameters: "
```
or log the actual values, removing the len?

Reply:
maybe in debug mode